Glossary page

OleanjiKingCode opened the pull request:
# Glossary page UI (No real data yet)


## How should this be tested?
You can check this out by adding'/glossary' to the url of the preview

Just like in the Figma design, I did the glossary page UI with dummy data to just show how it would look like for now.
 
![image](https://user-images.githubusercontent.com/75235148/200240864-62e668cc-986c-41f1-b47d-1ea51c3b5733.png)
![image](https://user-images.githubusercontent.com/75235148/200240875-55c1fe66-f1b6-4218-86ec-1907531537ed.png)
![image](https://user-images.githubusercontent.com/75235148/200240878-0a9691eb-fee4-49ce-8514-be3e35bf82e6.png)
![image](https://user-images.githubusercontent.com/75235148/200240895-bac02dd2-3ed6-4ca9-995b-3a64402f2df0.png)





## Linked issues

 closes https://github.com/EveripediaNetwork/issues/issues/840

Simplypheyie96 reviewed (changes requested):
it's starting to make sense
the tags after search is not properly done. 
The search box isn't done well yet.
then spacing between letter headings and the text boxes aren't well done too.
<img width="862" alt="image" src="https://user-images.githubusercontent.com/99387231/199739095-730700d8-a310-4348-9ffb-4990d3a671c3.png">

The heading looks odd on larger screens
![image](https://user-images.githubusercontent.com/99387231/199739045-269f0912-0bc6-46d1-b6f7-9255b2eb9b47.png)
 I would suggest you go with the design where the header is centered and without the braindao logo .
 
it is getting there. would be done in no time at all

Simplypheyie96 reviewed (changes requested):
well done @OleanjiKingCode 
there are still things to work on
- The color of the tags are not right. it should be similar to the the tags on the homepage.
<img width="703" alt="image" src="https://user-images.githubusercontent.com/99387231/200179075-1cd15e40-35ac-45b8-94d6-bbf8c5b4e8e7.png">

- On selection of the tags, it should looks something like this
<img width="564" alt="image" src="https://user-images.githubusercontent.com/99387231/200180362-ee59ff93-32ac-42e9-a347-5430de32fc1a.png">

- Spacing between headers and content doesn't look like an adjustment was made to it
<img width="1232" alt="image" src="https://user-images.githubusercontent.com/99387231/200180077-1eb55f86-f8d8-4a19-bf41-6a47a7fa84ad.png">

- There needs to be more spacing between the content and footer too
<img width="1218" alt="image" src="https://user-images.githubusercontent.com/99387231/200180183-26140919-a755-46ad-86b6-25204ed4ed14.png">
Ps: The cards are clickable

kesar commented on src/services/glossary/queries.ts:
```diff
@@ -0,0 +1,26 @@
+import { gql } from 'graphql-request'
+
+export const GET_TAGS_BY_ID = gql`
```
dont we have already something to get tags by id ? I though we use something like that for search / tag page

OleanjiKingCode replied:
@kesar I just tried searching for tags and it's not working, 
![image](https://user-images.githubusercontent.com/75235148/200236035-23a7aa84-f37f-4230-b9eb-213f5bb56154.png)
and it doesn't seem like a bug

kesar replied:
Good catch w search. Yes, that's something we can improve in future 👍🏻 


we have a page that show all tags related to a term. you can easily find that in the code or clicking in any tag.

https://github.com/EveripediaNetwork/ep-ui/blob/main/src/services/wikis/queries.ts#L332

Please try to take ownership of the task and spend time looking for such things. You shouldnt request a new review without spending time researching, it looks like you are in a rush to deliver something that its incomplete.

kesar commented on src/pages/glossary/index.tsx:
```diff
@@ -0,0 +1,143 @@
+import Link from '@/components/Elements/LinkElements/Link'
+
+const Glossary: NextPage = () => {
+  // const { data: GlossaryWikis } = useGetTagsByIdQuery({ id: 'glossary' })
```
remember to allow people to use tag Glossary (check how others tags are done)

OleanjiKingCode replied:
Okay kesar

Softdev1 reviewed (commented):
The suggestions made by @kesar  and @Simplypheyie96  should be implemented and then it will be retested .

kesar commented on src/components/Glossary/GlossaryWikiCard.tsx:
```diff
@@ -0,0 +1,51 @@
+interface GlossaryWikiCardProps {
+  wikiId: string
+  title: string
+  brief: string
```
probably summary might fit better. We haven't use brief anywhere and we have already a var name summary

kesar commented:
looking promising 👍🏻 . its missing:
- remove dummy data
- add results from tag by id
- remove duplicated code

kesar commented on src/components/Glossary/GlossaryItems.tsx:
```diff
@@ -0,0 +1,49 @@
+          </Box>
+          <Flex justifyContent="center" mb="-40px" mt="30px">
+            <GlossaryWikiCard
+              title="Lorem"
```
remove dummy data

OleanjiKingCode commented:
> 

On it now sir

kesar commented:
🙏🏻 thx 🚀 

> > 
> 
> On it now sir

OleanjiKingCode commented:
The glossary page is empty because there are no glossary tagged wikis
remove dummy data (done)
add results from tag by id (done)
remove duplicated code (done)

kesar commented:
looking promising. no need to request my review if its not fully done unless you have particular questions about the code / design.

so far tags are invisible in normal mode and search doesnt seem to work, but besides that, everything else looks good

kesar commented:
also lets not forget to link it in the footer and i guess learn section. sync w UX about that

Simplypheyie96 reviewed (changes requested):
it not yet there.
the tags are not right in light mode and the stroke around it is wrong. check the tags section on homepage. it should look like that
<img width="844" alt="image" src="https://user-images.githubusercontent.com/99387231/200550327-8022649b-a8d0-416e-a8fd-bf338d3b5ddd.png">

OleanjiKingCode commented:
> looking promising. no need to request my review if its not fully done unless you have particular questions about the code / design.
> 
> so far tags are invisible in normal mode and search doesnt seem to work, but besides that, everything else looks good

Already working on them, i wanted to ask that those tags and search results where do they link to? is it wikis or it scrolls to the section of the page.

kesar commented:
great question.

I would filter the list to show the results. on click in tags it should fill the search bar and do the same

![image](https://user-images.githubusercontent.com/1288106/200562855-d0e41c89-6cf2-4093-972f-5a03661a4c4f.png)

kesar commented on package.json:
```diff
@@ -78,6 +78,7 @@
     "react-infinite-scroll-hook": "^4.0.3",
     "react-markdown": "^8.0.1",
     "react-redux": "^7.2.7",
+    "react-scroll": "^1.8.8",
```
we already have https://www.npmjs.com/package/react-scroll-into-view cant you use that?

OleanjiKingCode replied:
Yes i used that before but im removing it in the next commit i push, because the react-scroll is much easier to use than the scroll into view

kesar replied:
ahh I see :)

I think in other places we just use https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView that perhaps let us avoid to use external libs.

external libs makes bigger bundles and reduce performance, please try to see if you can achieve the scrolling without adding extra stuff. If it becomes too complex / not possible then lets go w one lib.

OleanjiKingCode replied:
Alright sir

OleanjiKingCode commented:
@kesar ive fixed the alphabet responsive problem  and now it remains

getting the common wikis from the db and also entering the text of the common wiki tag selected into the search bar

kesar commented:
> @kesar ive fixed the alphabet responsive problem and now it remains
> 
> getting the common wikis from the db and also entering the text of the common wiki tag selected into the search bar

haha thx, that was fast :P

Simplypheyie96 reviewed (changes requested):
Not yet satisfied with this part.
it needs work.
![image](https://user-images.githubusercontent.com/99387231/201622582-b049e50c-57c8-43d5-b071-2d72a46422cb.jpeg)

OleanjiKingCode commented:
> Not yet satisfied with this part. it needs work. ![image](https://user-images.githubusercontent.com/99387231/201622582-b049e50c-57c8-43d5-b071-2d72a46422cb.jpeg)

Ohh I'll do that...I thought I already fixed it thanks

kesar commented on package.json:
```diff
@@ -21,10 +21,11 @@
+    "@chakra-ui/styled-system": "^2.3.4",
     "@chakra-ui/theme-tools": "^2.0.0",
-    "@chakra-ui/toast": "^2.0.0",
+    "@chakra-ui/toast": "^4.0.1",
```
revert libs updates unless you can explain why you need to update them and you have tested all the implementations of toast / chakra-ui react

OleanjiKingCode replied:
yes

OleanjiKingCode commented:
I updated the chakraui-react package to be able to use the highlight function and it needs chakraui/toast 4.0.1 dep
![image](https://user-images.githubusercontent.com/75235148/201687224-0d312a0b-1966-4e34-8242-93cb412719c6.png)
![image](https://user-images.githubusercontent.com/75235148/201687271-fd75f924-d1a0-494e-baac-65c12bd60213.png)

kesar commented:
> chakraui-react



> I updated the chakraui-react package to be able to use the highlight function and it needs chakraui/toast 4.0.1 dep ![image](https://user-images.githubusercontent.com/75235148/201687224-0d312a0b-1966-4e34-8242-93cb412719c6.png) ![image](https://user-images.githubusercontent.com/75235148/201687271-fd75f924-d1a0-494e-baac-65c12bd60213.png)

didnt it have already Highlight in our version? I havent seen any changelog about that in 2.3.7

OleanjiKingCode commented:
> > chakraui-react
> 
> > I updated the chakraui-react package to be able to use the highlight function and it needs chakraui/toast 4.0.1 dep ![image](https://user-images.githubusercontent.com/75235148/201687224-0d312a0b-1966-4e34-8242-93cb412719c6.png) ![image](https://user-images.githubusercontent.com/75235148/201687271-fd75f924-d1a0-494e-baac-65c12bd60213.png)
> 
> didnt it have already Highlight in our version? I havent seen any changelog about that in 2.3.7

I tried using it but it didnt work

OleanjiKingCode commented:
> > chakraui-react
> 
> > I updated the chakraui-react package to be able to use the highlight function and it needs chakraui/toast 4.0.1 dep ![image](https://user-images.githubusercontent.com/75235148/201687224-0d312a0b-1966-4e34-8242-93cb412719c6.png) ![image](https://user-images.githubusercontent.com/75235148/201687271-fd75f924-d1a0-494e-baac-65c12bd60213.png)
> 
> didnt it have already Highlight in our version? I havent seen any changelog about that in 2.3.7

I dont think there's  an highlight component in the chakra ui V 2.0.0 but I'm still doing a research on that

OleanjiKingCode commented:
I just added real data from the popular tags wiki and also changed the limit of the glossary wikis....
 I'm going to review the code again, (it doesn't look optimized yet )  just wanted to see the results

OleanjiKingCode commented:
@kesar the version 2.0.0 of the chakra UI react doesn't have the highlight component, it was implemented in the v2.2.3 

![image](https://user-images.githubusercontent.com/75235148/201970141-9af36a2f-d804-4fec-be36-9178575fe653.png)

kesar commented:
damm then it makes sense to update the lib. lets make sure we fully test the whole site before merging since this can affect to any part. once you guys have been tested it. lets merge it

btw we migrated to nextjs13 so pull first main changes before testing to make sure page works fine w new nextjs 🚜🚜

Emmanueldmlr commented:
@OleanjiKingCode , Let me know when you want to work on this branch so as to prevent conflict. I'm currently working on the branch.

OleanjiKingCode commented:
> damm then it makes sense to update the lib. lets make sure we fully test the whole site before merging since this can affect to any part. once you guys have been tested it. lets merge it
> 
> btw we migrated to nextjs13 so pull first main changes before testing to make sure page works fine w new nextjs 🚜🚜



> @OleanjiKingCode , Let me know when you want to work on this branch so as to prevent conflict. I'm currently working on the branch.

Alright, when you are done just say

Emmanueldmlr commented:
@Simplypheyie96, I'm happy to jump on a call with you to review the UI changes I made. Here is the link to the updated page https://iq-wiki-jthg8q7i3-prediqt.vercel.app/glossary. Let me know when you are free for a review.

Simplypheyie96 commented:
> @Simplypheyie96, I'm happy to jump on a call with you to review the UI changes I made. Here is the link to the updated page https://iq-wiki-jthg8q7i3-prediqt.vercel.app/glossary. Let me know when you are free for a review.

Now, this looks more like it. Great work!

OleanjiKingCode commented:
> @OleanjiKingCode, can you review the update I made to be sure everything works correctly

okay

OleanjiKingCode commented:
> @OleanjiKingCode, can you review the update I made to be sure everything works correct

Yes, its looks great, thanks

Emmanueldmlr commented:
@OleanjiKingCode, I noticed we are using dummy data, what's the plan?

OleanjiKingCode commented:
> @OleanjiKingCode, I noticed we are using dummy data, what's the plan?

Where boss the data isnt dummy

Softdev1 reviewed (commented):
I noticed that a newly created wiki title is immediately found on the Suggested Popular tags. I think the TagPopular API needs to be revisited. 
Here is the image : 
![image](https://user-images.githubusercontent.com/32770520/202216857-c0123c2f-79a0-4a06-9565-7bb633418360.png)
![image](https://user-images.githubusercontent.com/32770520/202217176-f12bb31a-438f-4045-a023-28fa82b850d1.png)

Emmanueldmlr commented:
> I noticed that a newly created wiki title is immediately found on the Suggested Popular tags. I think the TagPopular API needs to be revisited. Here is the image : ![image](https://user-images.githubusercontent.com/32770520/202216857-c0123c2f-79a0-4a06-9565-7bb633418360.png) ![image](https://user-images.githubusercontent.com/32770520/202217176-f12bb31a-438f-4045-a023-28fa82b850d1.png)

@Softdev1, this is not related to this PR. Kindly sync up with @kesar.

Emmanueldmlr commented:
@OleanjiKingCode, I removed a code block that is not necessary...Kindly ensure it doesn't break anything...I have tested it though, I just want a form of confirmation from you.

kesar commented:
its somehow related because we dont filter by tags there. that list needs to be of wiki titles (glossary terms). So for now, lets make it an array of hardcoded terms ( wiki titles ) that its somewhere easy to modify.

the idea its that if you click in one, title goes to search, and filter happens to match the result

OleanjiKingCode commented:
Yes I've tested it out and it works @Emmanueldmlr thanks

Emmanueldmlr commented:
@kesar @Simplypheyie96 @Softdev1 @OleanjiKingCode , I guess we need to merge this since @Softdev1 already created another issue to the next iteration.

Simplypheyie96 commented:
> @kesar @Simplypheyie96 @Softdev1 @OleanjiKingCode , I guess we need to merge this since @Softdev1 already created another issue to the next iteration.

yeah. This one is already too long

OleanjiKingCode commented:
Yeah I also thought about that.. but I was waiting for @kesar

kesar commented:
lfg
